### PR TITLE
✨♻️🐛 serviceslib[fastapi] and projects tables validation scripts 

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

- ✨ New diagnostics script:  in ``models-library/scripts`` a script to **validate db projects table** exported to a CSV
- ✨ Nex extra for ``servicelib[fastapi]`` package with **fastapi-only** features (released as **servicelib 1.1.0**)
   - adapts ``Makefile`` and ``CI`` 
   - utils to override ``FastApi.openapi``
   - extends ``functools``
- 🐛 FIXES ``openapi.json`` in ``catalog`` service
   - adds dependency to ``servicelib[fastapi]``
   - splits requirements ``_base.*, _packages.*``
- 🐛 FiXES pylint tests
- ♻️ cleanup ``pytest-simcore`` 

#### NOTES
- It is becoming very convoluted to deal with intra-dependencies using our current ``requirements``  approach. We should look for an simplified approach
- running ``models-library/scripts/validate-pg-projects.py`` revealed:
   - Found 7/1532 invalid projects in aws-production
   - Found 7/1535 invalid projects in aws-staging
   - Found 40/776 invalid projects in dalco-production
   - Found 12/1163 invalid projects in dalgo-staging

## Related issue/s

- ITISFoundation/osparc-issues#544: checks db validation errors in deployed environs
- ITISFoundation/osparc-issues#428

## How to test

- download CSV file for a projects table using Adminer export
- Let's place them in ``packages/models-library/scripts/ignore.keep`` and run a validation script
```bash
$ make devenv
$ source .venv/bin/activate
$ cd packages/models-library
$ make install-dev
$ cd scripts
$ mkdir ignore.keep
# place in scripts/ignore.keep all csv files
$ for f in ignore.keep/*.csv; do python validate-pg-projects.py -v $f >$f.log 2>&1 ; done
```
- open log files in ``ignore.keep/``

### CODE REVIEW

File: packages/service-library/Makefile [hunk 1]
```diff
@@ -4,41 +4,54 @@
...
 	@echo 'servicelib support for aiohttp extras is not included:'
-	@echo '- `make install-dev`: without aiohttp extras'
-	@echo '- `make "install-dev[aiohttp]"`: with aiohttp extras'
-    
-.PHONY: install-dev install-dev[aiohttp] install-ci install-ci[aiohttp]
-install-dev install-dev[aiohttp] install-ci install-ci[aiohttp]: _check_venv_active ## install app in development/production or CI mode
+	@echo '- `make install-dev`: common features w/o any extras'
+	@echo '- `make "install-dev[aiohttp]"`: w/ aiohttp extras'
+	@echo '- `make "install-dev[fastapi]"`: w/ fastapi extras'
+	@echo '- `make "install-dev[all]"`: w/ all extras'
```
Comment (GitHK) [inline packages/service-library/Makefile:17]: by all extras you mean both aiohttp and fastapi I guess?

File: packages/service-library/requirements/ci[aiohttp].txt [hunk 2]
```diff
@@ -15,4 +15,4 @@
 ../pytest-simcore/
 
 # current module
-.
+.[aiohttp]
```
Comment (GitHK) [inline packages/service-library/requirements/ci[aiohttp].txt:18]: 👍

File: packages/service-library/setup.py [hunk 3]
```diff
@@ -17,15 +18,14 @@ def read_reqs(reqs_path: Path):
...
-aiohttp_requirements = read_reqs(
-    here / "requirements" / "_aiohttp.in"
-)  # WEAK requirements
-
 
 test_requirements = read_reqs(
     here / "requirements" / "_test.txt"
 )  # STRONG requirements
 
+aiohttp_requirements = read_reqs(here / "requirements" / "_aiohttp.in")
```
Comment (GitHK) [inline packages/service-library/setup.py:26]: could you please add a comment explaining why the `*.in` files are included here?

File: scripts/common.Makefile [hunk 4]
```diff
@@ -53,9 +53,9 @@ MAKE_C := $(MAKE) --no-print-directory --directory
 #
 
 
-.PHONY: help
+.PHONY: hel%
```
Comment (GitHK) [inline scripts/common.Makefile:56]: is this % intended and what does it do?

File: services/catalog/setup.py [hunk 5]
```diff
@@ -56,7 +68,7 @@ def read_reqs(reqs_path: Path):
         ],
     },
     include_package_data=True,
-    install_requires=install_requirements,
+    # install_requires=install_requirements,
```
Comment (GitHK) [inline services/catalog/setup.py:71]: is this correct? should it stay commented?

**Review by GitHK — APPROVED** [review]
Cleanup looks very nice.

Below some minor stuff and possible concerns.

File: packages/service-library/Makefile (hunk 1 above)
Comment (pcrespov) [reply packages/service-library/Makefile:17]: yes, in the [setuptools jargon](https://hanxiao.io/2019/11/07/A-Better-Practice-for-Managing-extras-require-Dependencies-in-Python/), those are called *extras-require*

File: scripts/common.Makefile (hunk 4 above)
Comment (pcrespov) [reply scripts/common.Makefile:56]: This fixes a warning produced in the catalogs ``help`` Makefile recipe
This is how something like "inheritance" can be achieved with makefiles. Check https://mattandre.ws/2016/05/makefile-inheritance/ for details.

File: services/catalog/setup.py (hunk 5 above)
Comment (pcrespov) [reply services/catalog/setup.py:71]: good catch! I missed that!

File: services/catalog/openapi.json [hunk 6]
```diff
@@ -363,7 +363,7 @@
             "required": true,
             "schema": {
               "title": "User Id",
-              "exclusiveMinimum": 0.0,
+              "exclusiveMinimum": false,
```
Comment (sanderegg) [inline services/catalog/openapi.json:366]: this does not look good. this should be a number. something like ```0``` no ??

File: services/catalog/openapi.json [hunk 7]
```diff
@@ -1604,7 +1625,7 @@
         "properties": {
           "minHeight": {
             "title": "Minheight",
-            "exclusiveMinimum": 0.0,
+            "exclusiveMinimum": false,
```
Comment (sanderegg) [inline services/catalog/openapi.json:1628]: same here

**Review by sanderegg — CHANGES_REQUESTED** [review]
looks very neat!
maybe check the issues with ```exclusiveMinimum``` that looks suspicious

File: packages/service-library/setup.py (hunk 3 above)
Comment (pcrespov) [reply packages/service-library/setup.py:26]: Most of the info is in [requirements/python-dependencies.md)](https://github.com/ITISFoundation/osparc-simcore/blob/d9125d4e44d63a5413bac78ff7c547c3e6d37dd7/requirements/python-dependencies.md) but i will extend it.

Basically, the idea is that *libraries* shall have *weak* constraint requirements while *services* shall have *hard* constraints (i.e. frozen).  Weak requirements are in the ``*.in`` files while *hard* are in the ``*.txt`` files.

The main reason is that the former are shared among different services and provides more degrees of freedom. In the case of the services, those are final and there is no need for those degrees of freedom.

Libraries requirements are only frozen for testing (therefore ``test_requirements= .. txt`` in libraries setup).

File: services/catalog/openapi.json (hunk 6 above)
Comment (pcrespov) [reply services/catalog/openapi.json:366]: Actually bool is the right type according to openapi standard. That is the reason why i had to override the ``FastAPI.openapi`` memeber. See [comments and links here](https://github.com/ITISFoundation/osparc-simcore/blob/74a3f3c19def0059df1a3260b8e8419ce2b81c76/packages/service-library/src/servicelib/fastapi/openapi.py#L46)

File: services/catalog/openapi.json (hunk 7 above)
Comment (pcrespov) [reply services/catalog/openapi.json:1628]: see above

File: services/catalog/openapi.json (hunk 6 above)
Comment (sanderegg) [reply services/catalog/openapi.json:366]: hmm I don't think I agree with this.
this is an integer type. what does exclusiveMinimum: false mean?

Comment (sanderegg) [reply services/catalog/openapi.json:366]: The following properties are taken directly from the JSON Schema definition and follow the same specifications:
- exclusiveMinimum

Comment (pcrespov) [reply services/catalog/openapi.json:366]: Ah, I noticed that i missed one link. Check this

https://github.com/tiangolo/fastapi/issues/240

Comment (pcrespov) [reply services/catalog/openapi.json:366]: Indeed ... the patch i apply is not complete. But now it should be. 

See https://github.com/ITISFoundation/osparc-simcore/blob/97470c3f5278b5d6aa577b17f591a01475900e09/packages/service-library/tests/fastapi/test_openapi.py#L24

**Review by sanderegg — APPROVED** [review]
great!

File: packages/service-library/setup.py (hunk 3 above)
Comment (GitHK) [reply packages/service-library/setup.py]: I had a hunch it was that. Thanks for clearing it up

**Review by odeimaiz — APPROVED** [review]

**Pull request merged**